### ISSUE TITLE
[HOLD for payment 2024-07-25] [$250] Split- Unknown user profile loads infinitely when opening from split confirmation page offline

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#42841]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.77-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** applause internal team
**Slack conversation:**

## Action Performed:
1. Go to staging.new.expensify.com
2. Go offline.
3. Go to FAB > Split expense > Scan.
4. Upload a receipt.
5. Select an unknown account (non-existing account or never interacted with before account).
6. Proceed to confirmation page.
7. Click on the unknown account.

## Expected Result:
The user profile of the unknown account will not load infinitely.

## Actual Result:
The user profile of the unknown account loads infinitely.

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/38435837/e0e2bf2e-0478-4e27-b201-0afc2ecb4bd4

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a2f8d719d411a342</li>
        <li>Upwork Job ID: 1796338202260049920</li>
        <li>Last Price Increase: 2024-05-31</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 103030367</li>
<li>abzokhattab | Contributor | 103030368</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @jliexpensify</details>

**Comment by m-natarajan** [comment]
We think that this bug might be related to #vip-vsb

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Unknown user profile loads infinitely when opening from split confirmation page offline
### What is the root cause of that problem?
 This issue occurs because when selecting a participant it tries to load the profile data and since the network is offline it keeps loading till the network is back. 

I think we should disable clicking on a participant while the it has an optimistic data the same as we are doing in the submit request 
### What changes do you think we should make in order to solve the problem?


make the participant field disabled if the personal data is optimistic like we are doing here:

https://github.com/Expensify/App/blob/cda874413f8957849464b470e5f6cc2d3b358ea1/src/components/MoneyRequestConfirmationList.tsx#L601

we can make it consistent and share the same condition in the split and submit request: 

to do that we change define a shared function `isParticipantDisabled` 
```js
    const isParticipantDisabled = (participant: MoneyRequestConfirmationListItem) =>
        !participant.isInvoiceRoom && !participant.isPolicyExpenseChat && !participant.isSelfDM && ReportUtils.isOptimisticPersonalDetail(participant.accountID ?? -1);
```

then use it [here](https://github.com/Expensify/App/blob/cda874413f8957849464b470e5f6cc2d3b358ea1/src/components/MoneyRequestConfirmationList.tsx#L601) 
and add `isDisabled:isParticipantDisabled(participantOption)`


### Alternative solution 

or to make it simple we can just add `isDisabled: ReportUtils.isOptimisticPersonalDetail(participant.accountID ?? -1)` [here](https://github.com/Expensify/App/blob/cda874413f8957849464b470e5f6cc2d3b358ea1/src/components/MoneyRequestConfirmationList.tsx#L601)

**Comment by jliexpensify** [comment]
Hmm yeah, can repro this one, moving it to #vip-split

**Comment by PrinceMakavana** [comment]
### **Proposal**
---

### **Please re-state the problem that we are trying to solve in this issue.**

Unknown user profile loads infinitely when opening from split confirmation page offline

**### What is the root cause of that problem?**

When the user opens a profile that doesn't yet exist in personalDetails also Not on Main Database , the app will attempt to load the profile data, but the user is currently offline so the loading will show forever.

https://github.com/Expensify/App/blob/0cb2dc052f01b6a024ac872ad8bf628159fe7873/src/pages/ProfilePage.tsx#L248

Here hasAvatar & isLoading Both True Because user Data not valid or you can say that no data avaliable.


**### What changes do you think we should make in order to solve the problem?**

According to https://github.com/Expensify/App/issues/42552#issuecomment-2135445308 for the workspace participant, we should make the participant look disabled if it's not an expencify user , otherwise if we show then it cause problem for other steps like chat section *(from profile section use can able to navigate to 1:1 chat section).

So that Right Solution  Just Disable Participant look disabled if it's not an expencify user.

Solution : 

https://github.com/Expensify/App/blob/0cb2dc052f01b6a024ac872ad8bf628159fe7873/src/pages/ProfilePage.tsx#L95

convert This `details` variable in `useState` and Add `useEffect` for that `details` variable.

**Comment by PrinceMakavana** [comment]
Contributor details
Your Expensify account email: princemakavana61@gmail.com
Upwork Profile Link: [Upwork Link](https://www.upwork.com/freelancers/princemakavana)

**Comment by PrinceMakavana** [comment]
Not from expert agencies, but I am javascript expert 😄. I Just want to contribute in open source projects

**Comment by jliexpensify** [comment]
Bumping @rojiphil for reviews

**Comment by rojiphil** [comment]
Thanks for all your proposals. 

Proposals by @abzokhattab and @dominictb got the RCA correct i.e. clicking on the participant is enabled when the participant profile is optimistic. 
Not having a visually disabled participant is not a correct RCA as per @PrinceMakavana  [proposal](https://github.com/Expensify/App/issues/42841#issuecomment-2143289405).

@abzokhattab proposes to use a similar implementation from `submit expenses`  for `split expenses` which is correct. 
@dominictb [proposal](https://github.com/Expensify/App/issues/42841#issuecomment-2141399626) also proposes the same solution but additionally suggested renaming `isDisabled` to `isNonInteractive`. However, this is a minor enhancement which is already proposed in another issue. It is better to be handled there. Also, I do not find any other significant value-add to prefer this proposal.

@abzokhattab [proposal](https://github.com/Expensify/App/issues/42841#issuecomment-2139660967) LGTM as that was the first one to get to the correct RCA and solution 
🎀👀🎀 C+ reviewed

**Comment by rojiphil** [comment]
> Not from expert agencies, but I am javascript expert. I Just want to contribute in open source projects.

@PrinceMakavana As mentioned already [here](https://github.com/Expensify/App/issues/42841#issuecomment-2145463184), PR should be raised only after you are hired for the job. Please go through the contributing guidelines [here](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) for more details.

**Comment by rojiphil** [comment]
@dominictb Looks like the word `renaming` caused the confusion. Anyway, using `isNonInteractive` is within the scope of a separate issue. Let us not discuss that here. 

Coming to the selection, the selected proposal was the first to find out the root cause and propose a solution for the same. Your proposal only adds to it which we would have anyway figured out during the PR stage. This is the reasoning behind selecting the existing approved proposal.

**Comment by rojiphil** [comment]
@dominictb Not sure if I still follow you. Can you please share some details as to how the UX experience of the user would change by using `isNonInteractive` instead of `isDisabled`?

**Comment by rojiphil** [comment]
> Sure, let me share the details and videos tomorrow. Basically if we use `disabled`, it will show the "disabled" cursor which looks broken to the user. If we use `isNonInteractive`, it will show normal pointer/text cursor.

@dominictb Oh! Isn't that exactly what has been proposed in the proposal [here](https://github.com/Expensify/App/issues/42552#issuecomment-2130906454)? 
If so, doing that here will be unfair, especially since that proposal has already been selected.

**Comment by rojiphil** [comment]
@dominictb 
> > the selected proposal was the first to find out the root cause and propose a solution for the same
> 
> I'm not sure I'd agree with this because the solution and changes proposed there obviously doesn't work for the issue and reproduction steps mentioned in the OP. It can easily be validated that it doesn't work by testing it the proposal out.

It is not mandatory to have a complete solution during proposal selection stage. Also, the selected proposal was not wrong on the RCA or on the solution. But I agree that it missed making the change in one of the early returns which would have been easily identified at the PR stage. So, I don’t think it is fair to penalise the selected proposal just because it failed to make the change in one of the early returns. 

> I'm wondering how could it be figured out during PR when that prop doesn't exist yet in the code, and was only suggested in my proposal 😃

Please note that using `isDisabled` is in itself sufficient enough to resolve the problem here.
Further, while I agree that using `isInteractive` improvises over `isDisabled` as mentioned in another [issue](https://github.com/Expensify/App/issues/42552#issuecomment-2140729683), the difference is only concerning using `default` cursor over `disabled` cursor and I do not see this as a deciding factor in this issue. Also, once the improvements are made in that PR, we will have eventually figured out to replace the `isDisabled` added here while reviewing all existing `isDisabled` props.

So, I am still not convinced of your concerns. 
Leaving it to the internal engineer to decide.

**Comment by rojiphil** [comment]
@youssef-lr Post my review feedback [here](https://github.com/Expensify/App/issues/42841#issuecomment-2149378119), @dominictb had certain concerns [here](https://github.com/Expensify/App/issues/42841#issuecomment-2149701006) and [here](https://github.com/Expensify/App/issues/42841#issuecomment-2149861174). Also, my response is [here](https://github.com/Expensify/App/issues/42841#issuecomment-2151783015). 

However, I am not convinced of the raised concerns and am still inclined to go ahead with the selected proposal as per the review [here](https://github.com/Expensify/App/issues/42841#issuecomment-2149378119).

Not sure if I am missing something here but please weigh in here to take this forward.

**Comment by hayata-suenaga** [comment]
@rojiphil can you summarize what's going on here 😄 ?

**Comment by hayata-suenaga** [comment]
I just realized I'm not assigned to this issue (I just commented on this issue because I was assigned to the PR that one of the contributors opened without waiting for the proposal to be accepted)

I'll let @youssef-lr decide which proposal to go 🙇

**Comment by rojiphil** [comment]
@youssef-lr  Gentle bump to weigh in here. The summary is [here](https://github.com/Expensify/App/issues/42841#issuecomment-2154204865).

**Comment by jliexpensify** [comment]
Bump @youssef-lr for a review please!

**Comment by youssef-lr** [comment]
@abzokhattab's solution LGTM as well

**Comment by rojiphil** [comment]
* [x]  [@rojiphil] The PR that introduced the bug has been identified. Link to the PR: [Offending PR](https://github.com/Expensify/App/pull/40386)
* [x]  [@rojiphil] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: [Added comment](https://github.com/Expensify/App/pull/40386/files#r1690815153)
* [x]  [@rojiphil] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not Required. Existing checklist is good enough to capture such issues.
* [x]  [@rojiphil] Determine if we should create a regression test for this bug. :  Since this is an ui improvement, it is better to include regression tests 
* [x]  [@rojiphil] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. 

Test 1:
	1	Go to FAB > Split expense > Manual.
	2	Select an unknown account (non-existing account or never interacted with before account).
	3	Proceed to the confirmation page.
	4	Click on the unknown account.
	5	ensure that the participant row is not clickable and thus the user's profile page is not opened
Test 2:
	1	Go to FAB > Split expense > Scan.
	2	Upload a receipt.
	3	Select an unknown account (non-existing account or never interacted with before account).
	4	Proceed to the confirmation page.
	5	Click on the unknown account.
	6	ensure that the participant row is not clickable and thus the user's profile page is not opened

**Comment by jliexpensify** [comment]
I'll create a new upworks job, current one has expired.

**Comment by jliexpensify** [comment]
Payment Summary

- C: @abzokhattab $250
- C+: @rojiphil $250

[Have invited you both to the new Upworks job.](https://www.upwork.com/jobs/~0188a35285377fc931)

**Comment by rojiphil** [comment]
@jliexpensify Accepted the offer. Thanks.

**Comment by jliexpensify** [comment]
Thanks, paying now!

**Comment by jliexpensify** [comment]
Paid and job closed!
